cpp/src/join/mixed_join_kernels_semi.cu

```diff
@@ -70,11 +67,10 @@ __attribute__((visibility("hidden"))) __launch_bounds__(block_size) __global__
   if (outer_row_index < outer_num_rows) {
     // Figure out the number of elements for this key.
     auto equality = single_expression_equality<has_nulls>{
-      evaluator, thread_intermediate_storage, swap_tables, equality_probe};
+      evaluator, thread_intermediate_storage, false, equality_probe};
 
-    if ((join_type == join_kind::LEFT_ANTI_JOIN) !=
-        (hash_table_view.contains(outer_row_index, hash_probe, equality))) {
-      *(join_output_l + join_result_offsets[outer_row_index]) = outer_row_index;
+    if ((hash_table_view.contains(outer_row_index, hash_probe, equality))) {
```

Comment:
Remove the extra parentheses.
```suggestion
    if (hash_table_view.contains(outer_row_index, hash_probe, equality)) {
```

Reply:
Actually, could this be
```cpp
  *(raw_flagged + outer_row_index) = hash_table_view.contains(outer_row_index, hash_probe, equality);
```
to avoid thread divergence?

Reply:
Fixed, this also allows to remove the initialization kernel